### PR TITLE
feat(wallet-dashboard): polish homepage scroll

### LINKED ISSUE #4492: Polish homepage scroll
Improve homepage my managing the grid so that everything is visible in the viewport and coins and tx history have their own scrollbars

### PR DESCRIPTION
**Pull request by evavirseda**
# Description of change

Please write a summary of your changes and why you made them.

## Links to any relevant issues

fixes https://github.com/iotaledger/iota/issues/4492

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/wallet-dashboard/components/VirtualList.tsx [hunk 1]
```diff
@@ -51,17 +51,10 @@ function VirtualList<T>({
         if (lastItem.index >= items.length - 1 && hasNextPage && !isFetchingNextPage) {
             fetchNextPage();
         }
-    }, [
-        hasNextPage,
-        fetchNextPage,
-        items.length,
-        isFetchingNextPage,
-        virtualizer,
-        virtualizer.getVirtualItems(),
```
Comment (cpl121) [inline apps/wallet-dashboard/components/VirtualList.tsx]: why has this been removed from the useEffect?
Its still needed to get the last item

File: apps/wallet-dashboard/app/globals.css [hunk 2]
```diff
@@ -19,6 +19,7 @@ body {
 
 @layer components {
     .home-page-grid-container {
+        height: calc(100vh - 130px);
```
Comment (cpl121) [inline apps/wallet-dashboard/app/globals.css]: why 130px? I have a little scroll
Captura desde 2024-12-16 14-17-41

File: apps/wallet-dashboard/app/(protected)/home/page.tsx [hunk 3]
```diff
@@ -25,15 +25,15 @@ function HomeDashboardPage(): JSX.Element {
         <main className="flex flex-1 flex-col items-center space-y-8 py-md">
             {connectionStatus === 'connected' && account && (
                 <>
-                    <div className="home-page-grid-container h-full w-full">
+                    <div className="home-page-grid-container w-full">
```
Comment (cpl121) [inline apps/wallet-dashboard/app/(protected)/home/page.tsx]: Its weird for an account without tokens. All the sections should be together, right?
image

**Review by cpl121 — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/components/VirtualList.tsx (hunk 1 above)
Comment (evavirseda) [reply apps/wallet-dashboard/components/VirtualList.tsx]: I forgot to add the new one, sorry :sweat_smile:  :pray:

**Review by cpl121 — CHANGES_REQUESTED** [review]
NJ!
Only conflicts 🙏🏼

File: apps/wallet-dashboard/app/globals.css [hunk 4]
```diff
@@ -110,6 +110,7 @@ body {
 
     @screen md {
         .home-page-grid-container {
+            height: calc(100vh - 140px);
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/app/globals.css:101]: i would still make a min height so this doesnt happen 
image

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
looks great, just a nit

**Review by begonaalvarezd — APPROVED** [review]

**Review by brancoder — CHANGES_REQUESTED** [review]
we could add some padding under button segments so the scrolled coins dont disappear right under the segment
image

this is a nit about the segmented buttons and is probably not related to this PR but on smaller viewports the segmented button looks too rounded when it spans in to the second row 
image

**Review by cpl121 — APPROVED** [review]

**Review by brancoder — APPROVED** [review]
LGTM

**Pull request merged**